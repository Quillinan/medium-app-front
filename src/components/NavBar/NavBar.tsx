import UserMenu from '../UserMenu/UserMenu';

const NavBar: React.FC<{
  currentTab: string;
  onTabChange: (tab: string) => void;
}> = ({ currentTab, onTabChange }) => {
  const tabs = ['Dashboard', 'Team', 'Projects', 'Calendar', 'Reports'];

  return (
    <nav className='bg-gray-800'>
      <div className='mx-auto max-w-7xl px-4 sm:px-6 lg:px-8'>
        <div className='flex h-16 items-center justify-between'>
          <div className='flex items-center'>
            <div className='flex-shrink-0'>
              <img
                className='h-8 w-8'
                src='https://tailwindui.com/plus/img/logos/mark.svg?color=indigo&shade=500'
                alt='Your Company'
              />
            </div>
            <div className='hidden md:block'>
              <div className='ml-10 flex items-baseline space-x-4'>
                {tabs.map(tab => (
                  <button
                    key={tab}
                    onClick={() => onTabChange(tab)}
                    className={`rounded-md px-3 py-2 text-sm font-medium ${
                      currentTab === tab
                        ? 'bg-gray-900 text-white'
                        : 'text-gray-300 hover:bg-gray-700 hover:text-white'
                    }`}
                  >
                    {tab}
                  </button>
                ))}
              </div>
            </div>
          </div>
          <UserMenu />
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
