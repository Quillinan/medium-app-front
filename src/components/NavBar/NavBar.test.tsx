import { fireEvent, render, screen } from '@testing-library/react';
import NavBar from './NavBar';

vi.mock('@components/UserMenu/UserMenu', () => {
  return {
    default: () => <div>UserMenu</div>,
  };
});

describe('NavBar Component', () => {
  const onTabChangeMock = vi.fn();

  beforeEach(() => {
    onTabChangeMock.mockClear();
  });

  it('should render the logo', () => {
    render(<NavBar currentTab='Menu' onTabChange={onTabChangeMock} />);

    const logo = screen.getByAltText('Your Company');
    expect(logo).toBeInTheDocument();
  });
  it('should render the tabs', () => {
    render(<NavBar currentTab='Menu' onTabChange={onTabChangeMock} />);

    const tabs = ['Menu', 'Dashboard', 'Aniversários'];
    tabs.forEach(tab => {
      expect(screen.getByText(tab)).toBeInTheDocument();
    });
  });
  it('should render the UserMenu', () => {
    render(<NavBar currentTab='Menu' onTabChange={onTabChangeMock} />);

    const UserMenu = screen.getByText('UserMenu');
    expect(UserMenu).toBeInTheDocument();
  });
  it('should trigger onTabChange when a tab is clicked', () => {
    render(<NavBar currentTab='Menu' onTabChange={onTabChangeMock} />);

    const dashboardTab = screen.getByText('Dashboard');
    fireEvent.click(dashboardTab);
    expect(onTabChangeMock).toHaveBeenCalledWith('Dashboard');
  });
  it('should apply active class to the current tab', () => {
    render(<NavBar currentTab='Aniversários' onTabChange={onTabChangeMock} />);

    const activeTab = screen.getByText('Aniversários');
    expect(activeTab).toHaveClass('bg-gray-900 text-white');

    const inactiveTab = screen.getByText('Menu');
    expect(inactiveTab).toHaveClass(
      'text-gray-300 hover:bg-gray-700 hover:text-white'
    );
  });
  it('should match snapshot', () => {
    const { asFragment } = render(
      <NavBar currentTab='Menu' onTabChange={onTabChangeMock} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
