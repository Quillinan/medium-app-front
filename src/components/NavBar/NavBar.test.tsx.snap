// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar Component > should match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="bg-gray-800"
  >
    <div
      class="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex h-16 items-center justify-between"
      >
        <div
          class="flex items-center"
        >
          <div
            class="flex-shrink-0"
          >
            <img
              alt="Your Company"
              class="h-8 w-8"
              src="https://tailwindui.com/plus/img/logos/mark.svg?color=indigo&shade=500"
            />
          </div>
          <div
            class="hidden md:block"
          >
            <div
              class="ml-10 flex items-baseline space-x-4"
            >
              <button
                class="rounded-md px-3 py-2 text-sm font-medium bg-gray-900 text-white"
              >
                Menu
              </button>
              <button
                class="rounded-md px-3 py-2 text-sm font-medium text-gray-300 hover:bg-gray-700 hover:text-white"
              >
                Dashboard
              </button>
              <button
                class="rounded-md px-3 py-2 text-sm font-medium text-gray-300 hover:bg-gray-700 hover:text-white"
              >
                Anivers√°rios
              </button>
            </div>
          </div>
        </div>
        <div>
          UserMenu
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
